gcd/src/main/java/org/spine3/server/storage/datastore/DsFilters.java

```diff
@@ -0,0 +1,468 @@
+import org.spine3.server.entity.storage.CompositeQueryParameter;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Set;
+
+import static com.google.cloud.datastore.StructuredQuery.CompositeFilter.and;
+import static com.google.cloud.datastore.StructuredQuery.PropertyFilter.eq;
+import static com.google.cloud.datastore.StructuredQuery.PropertyFilter.ge;
+import static com.google.cloud.datastore.StructuredQuery.PropertyFilter.gt;
+import static com.google.cloud.datastore.StructuredQuery.PropertyFilter.le;
+import static com.google.cloud.datastore.StructuredQuery.PropertyFilter.lt;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Predicates.not;
+import static com.google.common.collect.FluentIterable.from;
+import static com.google.common.collect.Lists.newLinkedList;
+import static java.util.Collections.emptySet;
+import static java.util.Collections.singleton;
+import static org.spine3.client.CompositeColumnFilter.CompositeOperator.ALL;
+
+/**
+ * A utility for working with the Datastore {@linkplain Filter filters}.
+ *
+ * @author Dmytro Dashenkov
+ */
+final class DsFilters {
+
+    /**
+     * Matches the {@link CompositeQueryParameter} instances joined by
+     * the {@linkplain org.spine3.client.CompositeColumnFilter.CompositeOperator#ALL conjunctive}
+     * operator.
+     */
+    private static final Predicate<CompositeQueryParameter> isConjunctive =
+            new Predicate<CompositeQueryParameter>() {
+                @Override
+                public boolean apply(@Nullable CompositeQueryParameter input) {
+                    return input != null && input.getOperator() == ALL;
+                }
+            };
+
+    /**
+     * Matches the {@link CompositeQueryParameter} instances joined by
+     * the {@linkplain org.spine3.client.CompositeColumnFilter.CompositeOperator#EITHER disjunctive}
+     * operator.
+     */
+    private static final Predicate<CompositeQueryParameter> isDisjunctive = not(isConjunctive);
+
+    private DsFilters() {
+        // Prevent utility class fromm being initialized.
+    }
+
+    /**
+     * Converts the given {@link CompositeQueryParameter} instances into the Datastore
+     * {@link Filter}s.
+     *
+     * <p>The returned {@code Collection} contains the same predicate as the passed
+     * {@code Collection} of query parameters. The difference is in
+     * <ul>
+     *     <li>type: from Spine {@link CompositeQueryParameter} to Datastore {@link Filter};
+     *     <li>format: all the resulting {@link Filter}s are conjunctive (as Datastore does not
+     *         support disjunction) and are related with each other in the disjunctive way.
+     * </ul>
+     *
+     * <p><i>Example:</i>
+     * <p>Given query predicates {@code p1}, {@code p2}, {@code p3}, {@code p4}, {@code p5} passed
```

Comment:
Please allow an empty line before this `<p>`.

Also this example paragraph should be formatted better. It's a bit difficult to read ATM.